src/openzeppelin/tests/test_account.cairo

```diff
@@ -0,0 +1,330 @@
+
+    let set_public_key_selector = 0x2e3e21ff5952b2531241e37999d9c4c8b3034cccc89a202a6bf019bdf5294f9;
+
+    assert(account.get_public_key() == initial_public_key, 'Should get initial public key');
+
+    let mut calldata = ArrayTrait::new();
+    let new_public_key = 0x789789;
+    calldata.append(new_public_key);
+    let call = Call {
+        to: account.contract_address, selector: set_public_key_selector, calldata: calldata
+    };
+
+    calls.append(call);
+    let ret = account.__execute__(calls);
+
+    assert(account.get_public_key() == 0x789789, 'Should get new public key');
+
+    // Test return value
+    let mut call_retval = ret.at(0).span();
+    assert(call_retval.len() == 0, 'Should be an empty response');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_validate() {
+    let calls = ArrayTrait::new();
+    let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
+
+    assert(account.__validate__(calls) == starknet::VALIDATED, 'Should validate correctly');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_multicall() {
+    let account = setup_dispatcher(Option::Some(@SIGNED_TX_DATA()));
+    let erc20 = deploy_erc20(account.contract_address, 1000);
+    let recipient1 = contract_address_const::<0x123>();
+    let recipient2 = contract_address_const::<0x456>();
+    let mut calls = ArrayTrait::new();
+
+    let transfer_selector = 0x83afd3f4caedc6eebf44246fe54e38c95e3179a5ec9ea81740eca5b482d12e;
+
+    let mut calldata1 = ArrayTrait::new();
+    let amount1: u256 = 300;
+    calldata1.append(recipient1.into());
+    calldata1.append(amount1.low.into());
+    calldata1.append(amount1.high.into());
+    let call1 = Call {
+        to: erc20.contract_address, selector: transfer_selector, calldata: calldata1
+    };
+
+    let mut calldata2 = ArrayTrait::new();
+    let amount2: u256 = 500;
+    calldata2.append(recipient2.into());
+    calldata2.append(amount2.low.into());
+    calldata2.append(amount2.high.into());
+    let call2 = Call {
+        to: erc20.contract_address, selector: transfer_selector, calldata: calldata2
+    };
+
+    calls.append(call1);
+    calls.append(call2);
+    let ret = account.__execute__(calls);
+
+    // Assert that the transfers were successful
+    assert(erc20.balance_of(account.contract_address) == 200, 'Should have remaining');
+    assert(erc20.balance_of(recipient1) == 300, 'Should have transferred');
+    assert(erc20.balance_of(recipient2) == 500, 'Should have transferred');
+
+    // Test return value
+    let mut call1_serialized_retval = ret.at(0).span();
+    let mut call2_serialized_retval = ret.at(1).span();
+    let call1_retval = Serde::<bool>::deserialize(ref call1_serialized_retval);
+    let call2_retval = Serde::<bool>::deserialize(ref call2_serialized_retval);
+    assert(call1_retval.unwrap(), 'Should have succeeded');
+    assert(call2_retval.unwrap(), 'Should have succeeded');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_public_key_setter() {
```

Comment:
should we also test the getter? it's being used widely in tests but not as the object of the test